src/modules/slugs/slugs.php

```diff
@@ -326,10 +326,10 @@ public function query_vars( $query_vars ) {
             }
 
             // If req_uri is empty or if it is a request for ourself, unset error.
-            if ( empty( $request ) || $req_uri == $self || strpos( $_SERVER['PHP_SELF'], 'wp-admin/' ) !== false ) {
+            if ( empty( $request ) || $req_uri == $self || strpos( $_SERVER['PHP_SELF'], qtranxf_get_admin_base().'/' ) !== false ) {
                 unset( $_GET['error'] );
                 unset( $error );
-                if ( isset( $perma_query_vars ) && strpos( $_SERVER['PHP_SELF'], 'wp-admin/' ) !== false ) {
+                if ( isset( $perma_query_vars ) && strpos( $_SERVER['PHP_SELF'], qtranxf_get_admin_base().'/' ) !== false ) {
```

Comment:
@spleen1981 do you know why we use PHP_SELF here instead of REQUEST_URI? Is it fine if I rewrite it like this?
```
str_starts_with( $_SERVER['REQUEST_URI'], wp_admin_url() )
```

The whole `query_vars` function is a mess. I don't understand well the last check and I'm not sure the implementations of `qtranxf_get_admin_base()` work for all configurations (URL mode, ...).

Reply:
> why we use PHP_SELF here instead of REQUEST_URI?

I would leave as it is, as `REQUEST_URI` is before URL rewrites (i.e. it's as per the original call URL) while `PHP_SELF` is after (i.e. it's as seen by PHP, not necessarily the original call URL).
Anyway still you need to compare with `qtranxf_get_admin_base()` instead of the entire `admin_url()`.

>  I'm not sure the implementations of qtranxf_get_admin_base() work for all configurations 

as `admin_url()` is built using `site_url()` + a suffix, I guess it will always work the opposite to retrieve that suffix.

Reply:
All right, thanks for the answer. However I think the check is too "relaxed". We should not strip the slashes. A path should contain the slashes. By removing the slashes we could assume wrongly some other paths containing `wp-admin` partially (or the value that is customized).

Reply:
One problem though is that PHP_SELF is a path to the local script, so it could contain backslashes on Windows.

Reply:
Not entirely sure about the values we could get with PHP_SELF.
Imo we should also check the values at the beginning of the string, not at any position. We should not assume the website doesn't contain other sub-folders with that name. The previous checks were not good. I fixed this for the neutral paths but with HTTP_REFERER there's also the same question.

Reply:
Mmm on Linux PHP_SELF doesn't start with a slash apparently unlike REQUEST_URI (and therefore the path given in neutral path but that should be fixed now).
Let's precise better these remain questions to finalize this PR:
- slashes vs backslashes (Windows?) in PHP_SELF (I don't know well which values this can take)
- maybe more precise checks for PHP_SELF and HTTP_REFERER to avoid assumptions

Reply:
> We should not strip the slashes

agree, that's why we changed from `str_replace` to `trim` in `qtranxf_get_admin_base()`, in order to preserve possible inner slashes

> One problem though is that PHP_SELF is a path to the local script, so it could contain backslashes on Windows.

mmm I'm not sure, that would be applicable to `PATH` but not to `PHP_SELF`, which I think is obtained starting from the URL. At least with my windows XAMPP installation that is the situation.

Reply:
> agree, that's why we changed from str_replace to trim in qtranxf_get_admin_base(), in order to preserve possible inner slashes

That's good but strictly speaking not sufficient, by "strip" I also include "trim". Slashes _might_ be removed, but in that case additional care should be taken for the position of the search. Looking for any stripped/trimmed string is bug-prone as in theory it can appear anywhere in the path, at any position. That's why I would rather keep the slashes but PHP_SELF is not exactly as REQUEST_URI regarding the format of the path.

Reply:
Let's consider the fixes go beyond scope of the patch to generalize to custom `wp-admin`. The patch should now be pretty conservative because it retrieves the same value for a default config. We'll handle possible fixes separately.